### PR TITLE
Use new upload endpoint

### LINKED ISSUE #16: CLI command `post_file_to_channel` fails with `AttributeError`
Currently, quetz-client upload fails from CLI as the package path is handled as `pathlib.Path` although it is passed as `str`. More concretely, the following call:

```
quetz-client post_file_to_channel my-channel my-package.tar.gz
```

fails with

```
  File "quetz_client/client.py", line 167, in post_file_to_channel
    url = f"{self.url}/api/channels/{channel}/upload/{file.name}"
                                                      ^^^^^^^^^
AttributeError: 'str' object has no attribute 'name'
```

### PR DESCRIPTION
**Pull request by simonbohnen**
- Fix the usage of the new upload endpoint (also see https://github.com/mamba-org/quetz/pull/594)
- Implement quetz README script
  - The client is validated by implementing the steps from the [quetz README](https://github.com/mamba-org/quetz)
  - see the run [here](https://github.com/mamba-org/quetz-client/actions/runs/4206689367/jobs/7300508538)
- Add `quetz` version compatibilty notice to README
- Closes #16

### CODE REVIEW

File: src/quetz_client/client.py [hunk 1]
```diff
@@ -164,18 +164,25 @@ def yield_packages(
...
-        response.raise_for_status()
+        try:
+            response = self.session.post(
+                url=url,
+                data=body_bytes,
+                params=params,
+            )
+            response.raise_for_status()
+        except requests.HTTPError as e:
+            print(f"Could not upload {file_path.name}: {e}")
```
Comment (janjagusch) [inline src/quetz_client/client.py]: Why do we want to print instead of raise?

File: src/quetz_client/client.py [hunk 2]
```diff
@@ -164,18 +164,25 @@ def yield_packages(
             yield Package(**user_json)
 
     def post_file_to_channel(self, channel: str, file: Path, force: bool = False):
-        url = f"{self.url}/api/channels/{channel}/upload/{file.name}"
-        body = open(file, "rb")
+        file_path = Path(file)
+        url = f"{self.url}/api/channels/{channel}/upload/{file_path.name}"
+        body = open(file_path, "rb")
+        body_bytes = body.read()
```
Comment (janjagusch) [inline src/quetz_client/client.py]: Please use a context manager or the `pathlib.read_bytes` function.

https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_bytes

File: tests/conftest.py [hunk 3]
```diff
@@ -14,11 +14,11 @@
...
-    with requests.get(url, stream=True) as response:
-        with NamedTemporaryFile() as file:
-            with open(file.name, "wb") as fp:
-                shutil.copyfileobj(response.raw, fp)
-            yield Path(file.name)
+    xtensor_raw_download = requests.get(url, stream=True).raw
+    with NamedTemporaryFile() as file:
+        with open(file.name, "wb") as fp:
+            shutil.copyfileobj(xtensor_raw_download, fp)
+        yield Path(file.name)
```
Comment (janjagusch) [inline tests/conftest.py]: Any reason you prefer this style?

File: README.md [hunk 4]
```diff
@@ -1,9 +1,17 @@
 # quetz-client
 
-A Python client to interact with a Quetz server.
+A Python client to interact with a Quetz server. Compatible with `quetz>=0.4.0`.
```
Comment (janjagusch) [inline README.md]: Since we're switching to the new upload endpoint, we're not compatible with 0.4.0 anymore, right?

File: .github/workflows/test-readme.sh [hunk 5]
```diff
@@ -0,0 +1,22 @@
+set -e
```
Comment (janjagusch) [inline .github/workflows/test-readme.sh]: Might be useful for stuff like this, although I have never used it myself: https://docs.python.org/3/library/doctest.html

File: tests/dev_config.toml [hunk 6]
```diff
@@ -0,0 +1,14 @@
+[sqlalchemy]
```
Comment (janjagusch) [inline tests/dev_config.toml:1]: Why is this file necessary?

**Review by janjagusch — COMMENTED** [review]
Some questions, but generally I'm very excited to use the new endpoint.

File: README.md (hunk 4 above)
Comment (simonbohnen) [reply README.md]: Yes, paste mistake 👌🏼

File: tests/dev_config.toml (hunk 6 above)
Comment (simonbohnen) [reply tests/dev_config.toml:1]: It's used as the configuration for the quetz instance in the README script, see [here](https://github.com/mamba-org/quetz-client/blob/44a27e3cb8e9aa4672b62d4f375ce6579100ae96/.github/workflows/test-readme.sh#L7). The same file is also used by the live tests in #2.

File: tests/conftest.py (hunk 3 above)
Comment (simonbohnen) [reply tests/conftest.py]: precommit was complaining a bit, see [here](https://github.com/mamba-org/quetz-client/actions/runs/4206689364/jobs/7300508630#step:4:131)

File: .github/workflows/test-readme.sh (hunk 5 above)
Comment (simonbohnen) [reply .github/workflows/test-readme.sh]: It looks like doctest is only for testing Python code, not shell scripts 🤔 I moved the invocation of the script to the pytests, I like that better I think.

**Review by janjagusch — APPROVED** [review]
